optimum/habana/transformers/models/mpt/modeling_mpt.py

```diff
@@ -305,6 +307,13 @@ def prepare_inputs_for_generation(
                 input_ids = input_ids[:, remove_prefix_length:]
             else:
                 input_ids = torch.index_select(input_ids, 1, token_idx - 1)
+            # Converting back to tuples as it should be, so there's no type mismatch when calling graph
+            past_key_values = tuple([tuple(kv) for kv in past_key_values])
+        elif bucket_internal and token_idx is not None:
```

Comment:
I've compared this to the llama/qwen2 (below) and I think this line should be 



```suggestion
        elif (reuse_cache or bucket_internal) and token_idx is not None:
```

https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/models/qwen2/modeling_qwen2.py#L890C14-L890C72
https://github.com/huggingface/optimum-habana/blob/main/optimum/habana/transformers/models/llama/modeling_llama.py#L1117C9-L1117C73

Reply:
@imangohari1, thanks for review!

There's no `reuse_cache` support for MPT yet. That's why I didn't add it to the condition.

Reply:
historical context: reuse_cache came first, and then this change: https://github.com/huggingface/optimum-habana/pull/1028

PR 1028 removes the need for reuse_cache. So for new model optimizations I think it is fine to only make changes in line with PR1028, and leave out reuse_cache related changes.

Only in older models, where we already had reuse_cache code, we accomodate both